modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -149,11 +149,73 @@ object ConstructionServiceSuite extends MutableIOSuite with Checkers with Transa
         Transfer,
         none,
         AccountIdentifier(address, none).some,
-        Amount(AmountValue(1L), Currency(CurrencySymbol("DAG"), CurrencyDecimal(8L))).some
+        Amount(AmountValue(1L), currency.Currency(currency.CurrencySymbol("DAG"), currency.CurrencyDecimal(8L))).some
       )
 
       val accountIdentifiers = cs.getAccountIdentifiers(List(operation))
       expect.eql(none, accountIdentifiers)
     }
   }
+
+  test("returns operations for a valid signed transaction hex") { res =>
+    implicit val (sp, k) = res
+
+    val cs = ConstructionService.make[IO]()
+
+    def txs = (KeyPairGenerator.makeKeyPair[IO], KeyPairGenerator.makeKeyPair[IO]).tupled.flatMap {
+      case (srcKey, dstKey) =>
+        val srcAddress = srcKey.getPublic.toAddress
+        val dstAddress = dstKey.getPublic.toAddress
+        val txCount = PosInt(100)
+
+        generateTransactions(srcAddress, srcKey, dstAddress, txCount)
+          .flatMap(_.traverse { hashedTransaction =>
+            KryoSerializer[F]
+              .serialize(hashedTransaction.signed)
+              .map(Hex.fromBytes(_))
+              .map((_, hashedTransaction.hash))
+              .liftTo[IO]
+          })
+    }
+
+    txs
+      .flatMap(_.traverse {
+        case (hex, _) =>
+          cs.parseTransaction(hex, true)
+            .rethrowT
+            .map(result => expect.eql(Amount(AmountValue(-1L), currency.DAG).some, result.operations.head.amount))
+      })
+      .map(_.fold)
+  }
+
+  test("returns operations for a valid unsigned transaction hex") { res =>
+    implicit val (sp, k) = res
+
+    val cs = ConstructionService.make[IO]()
+
+    def txs = (KeyPairGenerator.makeKeyPair[IO], KeyPairGenerator.makeKeyPair[IO]).tupled.flatMap {
+      case (srcKey, dstKey) =>
+        val srcAddress = srcKey.getPublic.toAddress
+        val dstAddress = dstKey.getPublic.toAddress
+        val txCount = PosInt(100)
+
+        generateTransactions(srcAddress, srcKey, dstAddress, txCount)
+          .flatMap(_.traverse { hashedTransaction =>
+            KryoSerializer[F]
+              .serialize(hashedTransaction.signed.value)
+              .map(Hex.fromBytes(_))
+              .map((_, hashedTransaction.hash))
+              .liftTo[IO]
+          })
+    }
```

Comment:
I see this is duplicated part (DRY - don't repeat yourself). Please take it outside the test case to reuse it here and there

Reply:
Refactored.